src/basic/util.h

```diff
@@ -22,32 +22,35 @@ void in_initrd_force(bool value);
 
 int on_ac_power(void);
 
-static inline unsigned log2u64(uint64_t n) {
+#define LOG2U64(x) (x > 1 ? (unsigned) __builtin_clzll(x) ^ 63U : 0)
```

Comment:
double eval of `x`?

Reply:
It turns out that if we use a block ({ }) to avoid the double eval, this does not work as a compile-time constant anymore. I added a comment instead.

Reply:
You could write it like this:
```c
#define CONST_LOG2ULL(x) (x > 1 ? (unsigned) __builtin_clzll(x) ^ 63U : 0)
#define NONCONST_LOG2ULL(x) ({ int _x = x; _x > 1 ? (unsigned) __builtin_clzll(_x) ^ 63U : 0; })
#define LOG2ULL(x) __builtin_choose_expr(__builtin_constant_p(x), CONST_LOG2ULL(x), NONCONST_LOG2ULL(x))
```

Reply:
Well, OK. But it's only executed on constants… I think this is too complex to implement "just in case". If the need ever arises, we can use it.

Reply:
i was about to propose the name, i.e. a single macro tat covers both cases. Less chance of using incorrectly...